pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -17,151 +17,273 @@ limitations under the License.
 package pod
 
 import (
-	"reflect"
+	"fmt"
 	"testing"
 
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/assert"
+
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
-	podtest "k8s.io/kubernetes/pkg/kubelet/pod/testing"
 	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
 )
 
-// Stub out mirror client for testing purpose.
-func newTestManager() (*basicManager, *podtest.FakeMirrorClient) {
-	fakeMirrorClient := podtest.NewFakeMirrorClient()
-	manager := NewBasicPodManager().(*basicManager)
-	return manager, fakeMirrorClient
-}
-
-// Tests that pods/maps are properly set after the pod update, and the basic
-// methods work correctly.
-func TestGetSetPods(t *testing.T) {
-	mirrorPod := &v1.Pod{
+var (
+	mirrorPod = &v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
-			UID:       "987654321",
-			Name:      "bar",
-			Namespace: "default",
+			UID:       types.UID("mirror-pod-uid"),
+			Name:      "mirror-static-pod-name",
+			Namespace: metav1.NamespaceDefault,
 			Annotations: map[string]string{
 				kubetypes.ConfigSourceAnnotationKey: "api",
 				kubetypes.ConfigMirrorAnnotationKey: "mirror",
+				kubetypes.ConfigHashAnnotationKey:   "mirror",
 			},
 		},
 	}
-	staticPod := &v1.Pod{
+	staticPod = &v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
-			UID:         "123456789",
-			Name:        "bar",
-			Namespace:   "default",
+			UID:         types.UID("static-pod-uid"),
+			Name:        "mirror-static-pod-name",
```

Comment:
name the same as for mirror. Seems intentional. If so - either have a constant defined or add a comment that those should match

Reply:
done. thanks for this!